### ISSUE TITLE
Fig slows down typing speed

### ISSUE THREAD

**Issue by anthwinter** [withfig/fig#625]
### Description:
> Please include a detailed description of the issue (and an image or screen recording, if applicable)

When fig is enabled in my .zshrc file, it makes my terminal laggy. When typing it is very slow to bring up the response.

I've included a video showing that if I comment out the fig config in .zshrc it works as it should and is responsive.

https://user-images.githubusercontent.com/4440999/143033776-bf9c93af-c51a-402c-92f0-4786f11d564d.mp4

### Details:
|macOS|Fig|Shell|
|-|-|-|
|12.0.1|Version 1.0.52 (B317)|/bin/zsh|
<details><summary><code>fig diagnostic</code></summary>
<p>
<pre>Version 1.0.52 (B317) [British – PC] 
UserShell: /bin/zsh
Bundle path: /Applications/Fig.app
Autocomplete: true
Settings.json: true
CLI installed: true
CLI tool path: /Users/awinter/.fig/bin/fig
Accessibility: true
Number of specs: 207
SSH Integration: false
Tmux Integration: true
Keybindings path: /Users/awinter/.fig/user/keybindings
iTerm Integration: true 
Hyper Integration: false
VSCode Integration: true
Docker Integration: false
Symlinked dotfiles: false
Only insert on tab: false
Installation Script: true
PseudoTerminal Path: /Users/awinter/.nvm/versions/node/v14.17.0/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/awinter/.fig/bin
SecureKeyboardInput: false
SecureKeyboardProcess: <none>
Current active process: /bin/zsh (25364) - /dev/ttys003
Current working directory: /Users/awinter
Current window identifier: 420/% (com.microsoft.VSCode)</pre>
</p>
</details>

**Comment by brendanfalk** [comment]
Yes this is definitely an issue we are aware of. We appear to be running some command synchronously vs async and that is the cause of the issue.

@anthwinter Out of interest
1. what are the specs (RAM and CPU) of your mac? 
2. What other shell plugins do you have? Would you be open to uploading your dotfiles here?

I am on m1 and the worst I get when I hold down enter is this:
Fig in dotfiles
![image](https://user-images.githubusercontent.com/4949076/144513886-9a3b5ed3-2305-46f6-933a-16389badcf41.png)

Fig commented out in dotfiles
![CleanShot 2021-12-02 at 13 14 50](https://user-images.githubusercontent.com/4949076/144514061-076a97bd-e7cb-458c-9ea0-b139269cc8f3.png)

**Comment by anthwinter** [comment]
Hey @brendanfalk 
CPU is 2 GHz Quad-Core Intel Core i5
RAM is 16 GB 3733 MHz LPDDR4X

I run Oh my zsh but have tried to uninstall that completely too and still had problems.

Are there any particular dotfiles you'd like to see? here is my .zshrc

```
#### FIG ENV VARIABLES ####
# Please make sure this block is at the start of this file.
[ -s ~/.fig/shell/pre.sh ] && source ~/.fig/shell/pre.sh
#### END FIG ENV VARIABLES ####

export ZSH=$HOME/.oh-my-zsh

ZSH_THEME="robbyrussell"

plugins=(zsh-z zsh-autosuggestions)

source $ZSH/oh-my-zsh.sh

export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion

#### FIG ENV VARIABLES ####
# Please make sure this block is at the end of this file.
[ -s ~/.fig/fig.sh ] && source ~/.fig/fig.sh
#### END FIG ENV VARIABLES ####
```

**Comment by brendanfalk** [comment]
We have seen another user face this issue on an old work mac which was running Docker.

My hunch is this issue will be fixed when we change the way we do our integration with zsh which we are pushing next week. Stay tuned!

**Comment by SeparateRecords** [comment]
Can we confirm if this issue is fixed on v1.0.55?

**Comment by anthwinter** [comment]
Yeah, this is doing much better for me! Happy to enable Fig again :) Thanks
